administrator/components/com_content/controllers/article.php

```diff
@@ -154,8 +155,35 @@ public function batch($model = null)
 	 *
 	 * @since	3.1
 	 */
-	protected function postSaveHook(JModelLegacy $model, $validData = array())
+	protected function postSaveHook(JModelLegacy $model, $validData)
 	{
+		JModelLegacy::addIncludePath(JPATH_ADMINISTRATOR . '/components/com_menus/models', 'MenusModel');
+		$itemModel = JModelAdmin::getInstance('Item', 'MenusModel');
+		$itemModel->addTablePath(JPATH_ADMINISTRATOR . '/components/com_menus/tables');
+
+		$componentId = JComponentHelper::getComponent('com_content')->id;
+
+		JLoader::register('ContentHelper', JPATH_ADMINISTRATOR . '/components/com_content/helpers/content.php');
+		$articleId = ContentHelper::getarticleid($validData['title']);
```

Comment:
In my opinion it is not good idea to get article id from `title`. Imagine you have two article which has same but in different category. For example: `red` in `mobile cover` and `cloth` category.

You must get id which is just inserted and you can get it using,
```php
$articleId = $model->getState($this->context . '.id');
```